extension/intents/slideshow/contentScript.js

```diff
@@ -0,0 +1,347 @@
+
+    // iterate thru list and add to array
+
+    mediaElements.forEach(element => {
+      // strip their current classes
+
+      element.className = "";
+      switch (element.tagName) {
+        case "IMG": {
+          const img = iframeDoc.createElement("img");
+          img.setAttribute("src", element.getAttribute("src"));
+          slideElements.push(img);
+
+          break;
+        }
+        case "VIDEO": {
+          const vid = iframeDoc.createElement("video");
+          vid.setAttribute("src", element.getAttribute("src"));
+          vid.setAttribute("controls", true);
+          if (element.hasAttribute("poster")) {
+            vid.setAttribute("poster", element.getAttribute("poster"));
+          }
+          if (element.hasChildNodes()) {
+            element.childNodes.forEach(function(childElement) {
+              if (childElement.tagName === "SOURCE") {
+                const srcElement = iframeDoc.createElement("source");
+                srcElement.setAttribute("src", childElement.src);
+                if (childElement.hasAttribute("type")) {
+                  srcElement.setAttribute("type", childElement.type);
+                }
+
+                vid.appendChild(srcElement);
+              }
+            });
+          }
+          slideElements.push(vid);
+
+          break;
+        }
+        case "IFRAME": {
+          if (isVideoSourceUrl(element.getAttribute("src"))) {
+            const iframe = iframeDoc.createElement("iframe");
+            iframe.setAttribute("src", element.getAttribute("src"));
+            iframe.setAttribute("width", document.documentElement.clientWidth);
+            iframe.setAttribute(
+              "height",
+              document.documentElement.clientHeight
+            );
+            slideElements.push(iframe);
+          }
+
+          break;
+        }
+      }
+    });
+
+    return slideElements;
+  }
+
+  function previousSlide() {
+    moveSlide(-1);
+  }
+
+  function nextSlide() {
+    moveSlide(1);
+  }
+
+  function moveSlide(steps) {
+    let newIndex = currentSlideIndex + steps;
+    if (newIndex > slideElements.length - 1) {
+      newIndex = 0;
+    }
+    if (newIndex < 0) {
+      newIndex = slideElements.length - 1;
+    }
+
+    showSlide(newIndex);
+  }
+
+  function showSlide(slideIndex) {
+    currentSlideIndex = slideIndex;
+    swapSlide(slideElements[slideIndex]);
+  }
+
+  function swapSlide(newSlide) {
+    if (slideContainer.hasChildNodes()) {
+      slideContainer.firstChild.remove();
+    }
+    if (newSlide) {
+      if (newSlide.tagName === "IFRAME") {
+        // enable iframe to resize first
+        newSlide.setAttribute("width", document.documentElement.clientWidth);
+        newSlide.setAttribute("height", document.documentElement.clientHeight);
+      }
+      slideContainer.appendChild(newSlide);
+    }
+  }
+
+  communicate.register("openSlide", async message => {
+    revealSlideshow();
```

Comment:
Right now it starts on the overview, but I think showing the first slide would be clearer.

Also if you could catch ArrowLeft and ArrowRight and go forward/back on slides it would be helpful. I probably was confused because there was side decoration over the buttons the first time I tried it, but I naturally expected keys to work.

Reply:
Made first slide the default view. Added Support for Arrow keys and escape key